### PR TITLE
Allow send XML or raw data with Apex

### PR DESCRIPTION
**Pull request by LiborVilimekMassive**
I need to make this request:

POST /services/apexrest/KeyDate

``` 
<?xml version="1.0" encoding="UTF-8"?>
<callout>
    <parameter name="AccountID">{Zuora account Id}</parameter>
</callout>
```

If JSON.stringify is forced and content-type as well, I cannot do this. However with the update above, this call will go through:

```
    const body = `<?xml version="1.0" encoding="UTF-8"?>
    <callout>
        <parameter name="AccountID">${subscription.accountId}</parameter>
     </callout>`;

    return connection.apex.post('/KeyDate/', body, { headers: { 'Content-Type': 'application/xml' } });
```

### CODE REVIEW

**Comment by LiborVilimekMassive** [comment]
PS: I would prefer to have some fields in options like "sendBodyRaw, dontChangeContentType", but not sure if you want to add fields like that

File: lib/api/apex.js [hunk 1]
```diff
@@ -36,12 +36,15 @@ Apex.prototype._createRequestParams = function(method, path, body, options) {
...
   }
-  if (!/^(GET|DELETE)$/i.test(method)) {
+  if (!/^(GET|DELETE)$/i.test(method) && !_headers["Content-Type"]) {
     _headers["Content-Type"] = "application/json";
   }
   params.headers = _headers;
   if (body) {
-    params.body = JSON.stringify(body);
+    params.body = body;
+    if ('object' === typeof options['body']) {
```
Comment (stomita) [inline lib/api/apex.js]: Maybe
```javascript
if (typeof body === 'object')
```

Comment (LiborVilimekMassive) [reply lib/api/apex.js]: @stomita - thanks, fixed

**Comment by LiborVilimekMassive** [comment]
@stomita - fixed

**Comment by cristiand391** [comment]
closing this b/c v1 is now deprecated

**Pull request closed without merging**